students/rhamersky/Lesson_4/mailroom_part2.py

```diff
@@ -0,0 +1,335 @@
+        print()
+
+def choice1():
+    '''
+    Calls a method to add donors and/or donations to the master donor tuple table when the user inputs '1'.
+    :return:
+    '''
+    # User selects option 1. Can display a list of the donors and where the user can add donors and donations.
+    print()  # --> Formatting
+    sub_menu.send_thankyou()
+
+def choice2():
+    '''
+    Calls a method to create a summary report for the user when the user inputs '2'.
+    :return:
+    '''
+    # User selects option 2. Creates a donation summary for the user.
+    print()  # --> Formatting
+    try:
+        sub_menu.create_report()
+    except ValueError:
+        print("There are no donors on the list to create a report.")
+
+def choice3():
+    '''
+    Calls a method to write and save emails to donors when the user inputs '3'.
+    :return:
+    '''
+    print()  # --> Formatting
+    files.write_email(tup_donor_names)
+
+def choice4():
+    '''
+    Exits the program when the user inputs '4'.
+    :return: Returns 'exit' which is the flag word to breakout of the program.
+    '''
+    print()  # --> Formatting
+    return "exit"
+
+def lifetime_donations(donors):
+    '''
+    Creates a dictionary that stores the donor's name as the 'key' and the lifetime donation as the 'value'.
+    :param donors: Tuple table that stores all the donor's donation information.
+    :return: Returns the newly created dictionary with the donor's name:lifetime donation
+    '''
+    # Local variables used for this function.
+    lifetime_donations = int()
+    dict_donor_donations = {}
+    # Creates a dictionary (donor's name:lifetime donation)
+    for donor in donors:
+        for donation in range(2, len(donor)):
+            lifetime_donations += donor[donation]
+        lifetime_donations = "${:,.2f}".format(lifetime_donations)  # --> Formats the total amount for gifts
+        dict_donor_donations[donor[0]] = lifetime_donations
+        lifetime_donations = int()  # --> Resets lifetime donations.
+    return dict_donor_donations
+
+def main():
+    '''
+    This is the main function. This will run the full program.
+    :return: This method does not return anything. Return is just used to clearly show where this method ends.
+    '''
+    while True:
+        # User menu to update the priority list.
+        print()  # --> Formatting
+        print("\t\tMenu of Options\n\n"
+              "\t\t1) Send a Thank You\n\n"
+              "\t\t2) Create a Report\n\n"
+              "\t\t3) Send letters to everyone\n\n"
+              "\t\t4) Quit\n")
+        # Error handles user's option choice.
+        global int_option
+        int_option = menu.menu_options()
+        if menu_dispatch[int_option]() == "exit":
+            break
+        else:
+            continue
+    return
+
+# -----Dispatch Dictionary Section-----
+menu_dispatch = {1 : choice1,
+                 2 : choice2,
+                 3 : choice3,
+                 4 : choice4
```

Comment:
it is better to name those functions with the actual action instead of numbered for readability. ex:
```
def send_thank_you: pass
def create_report: pass
def send_letters: pass
```

Reply:
That makes sense, updated functions for readability.